src/server/routes.pl

```diff
@@ -3675,23 +3675,23 @@
                        user: Username
                    }).
 authenticate(System_Askable, Request, Auth) :-
-    user_forwarded_header(Functor),
-    Term =.. [Functor, Username],
+    insecure_user_header(Header_Key),
+    Term =.. [Header_Key, Username],
```

Comment:
What happens if `Header_Key` is...?
* `''`
* `' '`
* `_`
* `.`
* something else I'm probably not thinking of

Looking at `=../2` in `swipl`, it seems to just use these as atoms directly. However, I don't think it would be good to rely on that, and I don't think we should expect any values like these.

Should there be restrictions on what `TERMINUSDB_FORWARDED_USER_HEADER` can be? I think it would be reasonable to accept a **string matching the regular expression `[A-Za-z0-9\-]+`**. (Note that it must be non-empty due to `+`.) We can always expand on that if, for some reason, it is needed.

Reply:
That makes sense

Reply:
Swipl complains already when you have very weird HTTP headers, example with single quotes containing header key:

```

Illegal HTTP parameter: X-'Forwarded User': admin (in_http_request)</p>
```